specs/interop/superchain-config.md

```diff
@@ -0,0 +1,126 @@
+# Superchain Configuration Interop
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Pausability](#pausability)
+- [Storage Layout](#storage-layout)
+- [Interface and properties](#interface-and-properties)
+  - [`addDependency`](#adddependency)
+  - [`authorizedPortals`](#authorizedportals)
+  - [`dependencySet`](#dependencyset)
+  - [`dependencySetSize`](#dependencysetsize)
+- [Events](#events)
+  - [`DependencyAdded`](#dependencyadded)
+- [Invariants](#invariants)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `SuperchainConfig` contract is extended to manage the dependency set and portal authorization
+for interoperability features.
+
+The contract is updated with a new `CLUSTER_MANAGER` role that has the ability to add chains to the dependency set.
+This role is handled similar to other roles in `SuperchainConfig`. It uses a `CLUSTER_MANAGER_SLOT` to store the address.
+
+### Pausability
+
+When the Pause is activated, the following methods are disabled:
+
+1. `SharedLockbox.unlockETH()`
+
+## Storage Layout
+
+The contract uses the following storage layout for interoperability features:
+
+```solidity
+struct SuperchainConfigDependencies {
+    ISharedLockbox sharedLockbox;
+    EnumerableSet.UintSet dependencySet;
+    mapping(address => bool) authorizedPortals;
+}
+```
+
+The `SuperchainConfigDependencies` struct contains the core state variables needed for managing interoperability
+between chains in the dependency set:
+
+- `sharedLockbox`: A reference to the `SharedLockbox` contract that manages unified ETH liquidity across the dependency set.
+  This allows ETH withdrawals to be processed from any chain's portal regardless of where the ETH was originally deposited.
+
+- `dependencySet`: An enumerable set containing the chain IDs of all chains in the dependency set.
+  This set is used to track which chains are part of the interoperable cluster.
+  The set is append-only - chains cannot be removed once added.
+
+- `authorizedPortals`: A mapping that tracks which `OptimismPortal` contracts are authorized to interact
+  with the `SharedLockbox`.
+  When a chain is added to the dependency set, its portal is authorized to lock and unlock ETH from
+  the shared liquidity pool.
+
+## Interface and properties
```

Comment:
we should really consider a function to see if a given chainId is within the dependency set